templates/example_rule.yml

```diff
@@ -0,0 +1,29 @@
+AnalysisType: rule
```

Comment:
Too lazy to convert it to YAML, but here are the list of fields you can use:

```
        "AnalysisType": Or("rule", "scheduled_rule"),
        "Enabled": bool,
        "Filename": str,
        "RuleID": And(str, NAME_ID_VALIDATION_REGEX),
        Or("LogTypes", "ScheduledQueries"): [str],
        "Severity": Or("Info", "Low", "Medium", "High", "Critical"),
        Optional("Description"): str,
        Optional("DedupPeriodMinutes"): int,
        Optional("DisplayName"): And(str, NAME_ID_VALIDATION_REGEX),
        Optional("OutputIds"): [str],
        Optional("Reference"): str,
        Optional("Runbook"): str,
        Optional("SummaryAttributes"): [str],
        Optional("Suppressions"): [str],
        Optional("Threshold"): int,
        Optional("Tags"): [str],
        Optional("Reports"): {str: list},
        Optional("Tests"): [
            {
                "Name": str,
                Optional(
                    "LogType"
                ): str,  # Not needed anymore, optional for backwards compatibility
                "ExpectedResult": bool,
                "Log": object,
            }
        ],
```

I would update this to indicate which ones are optional, and to include all the fields (e.g. I see `DedupPeriodMinutes` and `Threshold` are missing, important ones).

Or perhaps two template files, a minimal one with only required fields and a maximal one with all possible fields. They can both refer to the same python file.